### PR TITLE
Refactor `stay_prob` to `switch_prob` in `ExplorationWrapper`

### PR DESCRIPTION
**Pull request by kmdanielduan**
This PR aims to address #410. It refactors `stay_prob` to `switch_prob` in `ExplorationWrapper`.

### CODE REVIEW

File: tests/policies/test_exploration_wrapper.py [hunk 1]
```diff
@@ -57,11 +57,46 @@ def test_switch():
     assert num_constant > 450
 
 
+def test_switch_prob():
+    # Ensure that we test both the random and the wrapped policy
```
Comment (AdamGleave) [inline tests/policies/test_exploration_wrapper.py:61]: Meaning of comment unclear. Code block immediately following it checks that the policy is never switched (when `switch_prob=0.0`) which does not seem to connect to this.

Comment (AdamGleave) [inline tests/policies/test_exploration_wrapper.py:61]: Meaning of comment unclear. Code block immediately following it checks that the policy is never switched (when `switch_prob=0.0`) which does not seem to connect to this.

File: tests/policies/test_exploration_wrapper.py [hunk 2]
```diff
@@ -57,11 +57,46 @@ def test_switch():
...
+    wrapper, venv = make_wrapper(random_prob=0.5, switch_prob=0.0)
+    policy = wrapper.current_policy
+    np.random.seed(0)
+    obs = np.random.rand(100, 2)
+    for action in wrapper(obs):
+        assert venv.action_space.contains(action)
+        assert wrapper.current_policy == policy
+
+    for random_prob in [0.0, 0.5]:
+        wrapper, venv = make_wrapper(random_prob=random_prob, switch_prob=1.0)
```
Comment (AdamGleave) [inline tests/policies/test_exploration_wrapper.py:72]: I suggest factoring this code block out into a helper function, then calling it via something like:

```
num_random, num_constant = _wrapper_sample(random_prob=0.5, num_timesteps=1000, seed=0)
assert num_random > 450
assert num_constant > 450
num_random, num_constant = _wrapper_sample(random_prob=0.0, num_timesteps=1000, seed=0)
assert num_random == 0
assert num_constant == 1000
```

For loop seems a bit weird when we then have an if-else statement for every value the for loop can take.

File: src/imitation/policies/exploration_wrapper.py [hunk 3]
```diff
@@ -33,12 +33,12 @@ def __init__(
             policy: The policy to randomize.
             venv: The environment to use (needed for sampling random actions).
             random_prob: The probability of picking the random policy when switching.
-            stay_prob: The probability of staying with the current policy.
+            switch_prob: The probability of switch away from the current policy.
```
Comment (AdamGleave) [inline src/imitation/policies/exploration_wrapper.py:36]:
```suggestion
            switch_prob: The probability of switching away from the current policy.
```

**Review by AdamGleave — COMMENTED** [review]
Thanks for noticing and fixing this issue! Mostly looks good, some suggestion to clean up the tests, please request re-review once that's addressed.

File: tests/policies/test_exploration_wrapper.py (hunk 1 above)
Comment (kmdanielduan) [reply tests/policies/test_exploration_wrapper.py:61]: Yeah, I agree. Changed both tests into more detailed descriptions.

File: tests/policies/test_exploration_wrapper.py (hunk 2 above)
Comment (kmdanielduan) [reply tests/policies/test_exploration_wrapper.py:72]: Sounds good! I made corresponding changes to this.

**Comment by kmdanielduan** [comment]
Sounds good! I have made changes to clean up the tests.

File: tests/policies/test_exploration_wrapper.py [hunk 4]
```diff
@@ -68,33 +89,35 @@ def test_switch_prob():
...
-                else:  # pragma: no cover
-                    raise ValueError("Unknown policy")
-        if random_prob == 0.5:
-            # Holds with very high probability (and seeding means it's deterministic)
-            assert num_random > 450
-            assert num_constant > 450
-        else:
-            assert num_random == 0
-            assert num_constant == 1000
+            _ = wrapper(obs)
```
Comment (AdamGleave) [inline tests/policies/test_exploration_wrapper.py:99]:
```suggestion
            wrapper(obs)
```
It's OK to just discard return values in Python.

**Review by AdamGleave — APPROVED** [review]
LGTM

**Pull request merged**